File: app/src/main/java/com/example/covidnow/Article.java
package com.example.covidnow;

import android.text.format.DateUtils;

import org.json.JSONException;
import org.json.JSONObject;
import org.parceler.Parcel;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

@Parcel
public class Article {
    private String headline;
    private String source;
    private String date;
    private String summary;
    private String url;
    private String imageUrl;

    public Article() {};

    public Article(String headline, String source, String summary, String url, String date, String imageUrl) {
        this.headline = headline;
        this.source = source;
        this.date = date;
        this.summary = summary;
        this.url = url;
        this.imageUrl = imageUrl;
    }

    public static Article fromJson(JSONObject json) throws JSONException {
        Article article = new Article();
        article.headline = json.getString("title");
        article.summary = json.getString("excerpt");
        article.date = json.getString("publishedDateTime");
        article.source = json.getJSONObject("provider").getString("name");
        article.url = json.getString("webUrl");
        try {
            article.imageUrl = ((JSONObject) json.getJSONArray("images").get(0)).getString("url");
        } catch(Exception e) {
            article.imageUrl = null;
        }
        return article;
    }

    public String getHeadline() {
        return headline;
    }

    public String getSource() {
        return source;
    }

    public String getDate() {
        return date;
    }

    /*
    public String getStringDate() {
        return getRelativeTimeAgo(date);
    }*/

    public String getSummary() {
        return summary;
    }

    public String getUrl() {
        return url;
    }

    // getRelativeTimeAgo("Mon Apr 01 21:16:23 +0000 2014");
    public static String getRelativeTimeAgo(Date date) {
        String twitterFormat = "EEE MMM dd HH:mm:ss ZZZZZ yyyy";
        SimpleDateFormat sf = new SimpleDateFormat(twitterFormat, Locale.ENGLISH);
        sf.setLenient(true);

        String relativeDate = "";
        long dateMillis = date.getTime();
        relativeDate = DateUtils.getRelativeTimeSpanString(dateMillis,
                System.currentTimeMillis(), DateUtils.SECOND_IN_MILLIS).toString();
        return relativeDate;
    }

    public String getImageUrl() {
        return imageUrl;
    }
}

File: app/src/main/java/com/example/covidnow/models/Article.kt
package com.example.covidnow.models

import android.os.Parcelable
import android.text.format.DateUtils
import kotlinx.android.parcel.Parcelize
import org.json.JSONException
import org.json.JSONObject
import org.parceler.Parcel
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

@Parcel
@Parcelize
class Article() : Parcelable {
    var headline: String? = null
        private set
    var source: String? = null
        private set
    var date: String? = null
        private set
    var summary: String? = null
        private set
    var url: String? = null
        private set
    var imageUrl: String? = null
        private set

    companion object {
        @JvmStatic
        @Throws(JSONException::class)
        fun fromJson(json: JSONObject): Article {
            val article = Article()
            article.headline = json.getString("title")
            article.summary = json.getString("excerpt")
            var stringDate =  json.getString("publishedDateTime")
            article.date = getRelativeTimeAgo(stringDate)
            article.source = json.getJSONObject("provider").getString("name")
            article.url = json.getString("webUrl")
            try {
                article.imageUrl = (json.getJSONArray("images")[0] as JSONObject).getString("url")
            } catch (e: Exception) {
                article.imageUrl = null
            }
            return article
        }

        fun getRelativeTimeAgo(stringDate: String): String {
            // Change date to relative time
            val formatter: DateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
            var date = formatter.parse(stringDate) as Date
            // User twitter format date
            val format = "EEE MMM dd HH:mm:ss ZZZZZ yyyy"
            val sf = SimpleDateFormat(format, Locale.ENGLISH)
            sf.isLenient = true
            var relativeDate = ""
            val dateMillis = date.time
            relativeDate = DateUtils.getRelativeTimeSpanString(dateMillis,
                    System.currentTimeMillis(), DateUtils.SECOND_IN_MILLIS).toString()
            return relativeDate
        }
    }
}